carla/evaluation/redundancy.py

```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+from carla.models.api import MLModel
+
+
+def redundancy(
+    factuals: pd.DataFrame, counterfactuals: pd.DataFrame, mlmodel: MLModel
+) -> List[List[int]]:
+    """
+    Computes Redundancy measure for every counterfactual
+
+    Parameters
+    ----------
+    factuals: Encoded and normalized factual samples
+    counterfactuals: Encoded and normalized counterfactual samples
+    mlmodel: Black-box-model we want to discover
+
+    Returns
+    -------
+    List with redundancy values per counterfactual sample
+    """
+    redundancies = []
+
+    df_enc_norm_fact = factuals.reset_index(drop=True)
+    df_cfs = counterfactuals.reset_index(drop=True)
+
+    for idx, row in df_cfs.iterrows():
```

Comment:
I think here we can make good use of .apply() instead of iterrows

E.g. something like this `df_cfs["redundancies"] = df_cfs.apply(lambda x: compute_redundancy(...))`

Reply:
Never used `apply()`, will have a look into it 😄